aw_transform/filter_period_intersect.py

```diff
@@ -30,43 +61,96 @@ def filter_period_intersect(events: List[Event], filterevents: List[Event]) -> L
     Useful for example when you want to filter away events or
     part of events during which a user was AFK.
 
-    Example:
+    Usage:
       windowevents_notafk = filter_period_intersect(windowevents, notafkevents)
 
+    Example:
+      events1   |   =======        ======== |
+      events2   | ------  ---  ---   ----   |
+      result    |   ====  =          ====   |
+
     A JavaScript version used to exist in aw-webui but was removed in `this PR <https://github.com/ActivityWatch/aw-webui/pull/48>`_.
     """
 
-    events = sorted(events, key=lambda e: e.timestamp)
-    filterevents = sorted(filterevents, key=lambda e: e.timestamp)
-    filtered_events = []
+    events = sorted(events)
+    filterevents = sorted(filterevents)
 
-    e_i = 0
-    f_i = 0
-    while e_i < len(events) and f_i < len(filterevents):
-        event = events[e_i]
-        filterevent = filterevents[f_i]
-        ep = _get_event_period(event)
-        fp = _get_event_period(filterevent)
+    return [_replace_event_period(e1, ip) for (e1, _, ip) in _intersecting_eventpairs(events, filterevents)]
 
-        ip = ep.intersection(fp)
-        if ip:
-            # If events itersected, add event with intersected duration and try next event
-            filtered_events.append(_replace_event_period(event, ip))
-            if ep.end <= fp.end:
-                e_i += 1
-            else:
-                f_i += 1
+
+def period_union(events1: List[Event], events2: List[Event]) -> List[Event]:
+    """
+    Takes a list of two events and returns a new list of events covering the union
+    of the timeperiods contained in the eventlists with no overlapping events.
+
+    WARNING: This function strips all data from events as it cannot keep it consistent.
+
+    Example:
+      events1   |   =======       ========= |
+      events2   | ------  ---  --    ----   |
+      result    | -----------  -- ========= |
```

Comment:
Use all = or - in the example, gets confusing otherwise considering the warning

Reply:
I think the idea was to show which event will be stretched to the length of the other (or overwrite the overlapping event), but considering it strips all of the data it may be only confuthing in here.

Reply:
True, done!